### PR TITLE
feat(app): add thermocycler module card in device details

### LINKED ISSUE #9326: Device Details: TC Module Card
# Overview

This ticket outlines the frontend side side of the TC module card in device details.

# Implementation details
The module card should be composed out of a generic `ModuleCard` component that can be shared by all other modules

Conditionally render the TC module card inside of `app/src/pages/Devices/DeviceDetails/index.tsx` if there is a TC module attached.

# Design
module card design: https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=1864%3A117484

overflow settings design (at the top of this frame): https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=1297%3A91474

# Acceptance criteria
Module card should display...
- [x] A rendering of the TC module
- [x] The port the module is connected in
- [x] The TC icon
- [x] The module's display name
  - [x] The module lid status
- [x] The module lid target temperature (in Celcius). target temperature should read N/A if no target has been set
- [x] The module lid current temperature (in Celcius) 
- [x] The module's block target temperature (in Celcius). target temperature should read N/A if no target has been set
- [x] The module's block current temperature (in Celcius) 
- [ ] When clicking on the settings button, it should display two dropdown buttons to `Set lid temperature` and `Set block temperature`. For now, when each button is clicked `console.log` "set lid temperature" or "set block temperature"

### PR DESCRIPTION
**Pull request by sakibh**
# Overview

This PR implements the thermocycler module card. closes #9326

image

# Changelog

- Added `ThermocyclerModuleData` component to add TC Module Card.
- Added tests

# Review requests

- Check that the thermocycler card is rendering properly and meets the design specs of this [Figma node](https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=1297%3A87906).
- Ensure all the ACs in the ticket being addressed are met in this PR (Note: the last AC is not complete as it requires the overflow dropdown menu to be completed; a follow up PR will address that).

# Risk assessment

- Low

### CODE REVIEW

File: app/src/organisms/Devices/ModuleCard/ThermocyclerModuleData.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      case 'cooling':
+      case 'heating': {
+        StatusLabelProps.backgroundColor = C_SKY_BLUE
+        //  TODO IMMEDIATELY: animate iconColor
+        break
+      }
+      case 'error': {
+        StatusLabelProps.backgroundColor = '#fffcf5'
+        StatusLabelProps.iconColor = '#F09D20'
+        StatusLabelProps.textColor = '#7B5B09'
```
Comment (sakibh) [inline app/src/organisms/Devices/ModuleCard/ThermocyclerModuleData.tsx]: Will rebase/refactor whenever the PR with the new UI colors are added to the components library.

File: app/src/atoms/StatusLabel/index.tsx [hunk 2]
```diff
@@ -11,13 +11,14 @@ import {
   FONT_SIZE_CAPTION,
 } from '@opentrons/components'
 interface StatusLabelProps {
-  status: string
+  status: string | null
```
Comment (jerader) [inline app/src/atoms/StatusLabel/index.tsx]: why did you add `null`?

File: app/src/organisms/Devices/ModuleCard/ThermocyclerModuleData.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  C_DARK_GRAY,
+  DIRECTION_COLUMN,
+  SPACING_4,
+  C_SILVER_GRAY,
+  C_DARK_BLACK,
+  C_SKY_BLUE,
+  C_BLUE,
+  C_BLUE_PRESSED,
+} from '@opentrons/components'
+import { ThermocyclerStatus } from '../../../redux/modules/api-types'
```
Comment (jerader) [inline app/src/organisms/Devices/ModuleCard/ThermocyclerModuleData.tsx]: nit: you can add `type` here

File: app/src/organisms/Devices/ModuleCard/ThermocyclerModuleData.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+interface ThermocyclerModuleProps {
+  status: ThermocyclerStatus | null
+  currentTemp: number | null
```
Comment (jerader) [inline app/src/organisms/Devices/ModuleCard/ThermocyclerModuleData.tsx:29]: curious why the `currentTemp` can be null. But i see it's included in the type.

File: app/src/organisms/Devices/ModuleCard/ThermocyclerModuleData.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    status: string | null
+  ): { backgroundColor: string; iconColor: string; textColor: string } => {
+    const StatusLabelProps = {
+      backgroundColor: C_SILVER_GRAY,
+      iconColor: C_DARK_GRAY,
+      textColor: C_BLUE_PRESSED,
+    }
+
+    switch (status) {
+      case 'idle': {
```
Comment (jerader) [inline app/src/organisms/Devices/ModuleCard/ThermocyclerModuleData.tsx:52]: Can you add a few tests to `statusLabel.test` to test these different thermocycler statuses? I guess it will kind of get redundant once status label becomes more widely used but I think it is important to thoroughly test these different conditions.

File: app/src/organisms/Devices/ModuleCard/ThermocyclerModuleData.tsx (hunk 4 above)
Comment (shlokamin) [reply app/src/organisms/Devices/ModuleCard/ThermocyclerModuleData.tsx:29]: same with `status `

**Review by shlokamin — COMMENTED** [review]
it looks like the width of the cards is a little big compared to figma, is that just me?

File: app/src/organisms/Devices/ModuleCard/ThermocyclerModuleData.tsx (hunk 4 above)
Comment (sakibh) [reply app/src/organisms/Devices/ModuleCard/ThermocyclerModuleData.tsx:29]: For `status` it actually does not need the null type, I removed it. However, for `currentTemp` I was following the interface from `ThermocyclerData` in `api-types.ts`. I'm unable to test if it actually ever returns null in any scenario but TS is expecting it to have that type.

File: app/src/atoms/StatusLabel/index.tsx (hunk 2 above)
Comment (sakibh) [reply app/src/atoms/StatusLabel/index.tsx]: I removed it. TS was enforcing this because I had added a null type for status in `ThermocyclerModuleData` as well.

**Comment by sakibh** [comment]
> it looks like the width of the cards is a little big compared to figma, is that just me?

Yeah the cards are wider because it was set at 24rem. The Figma has the cards set at 317px which is ~20rem. I will update this.

File: app/src/organisms/Devices/ModuleCard/MagneticModuleData.tsx [hunk 6]
```diff
@@ -25,12 +26,12 @@ export const MagneticModuleData = (
   const { t } = useTranslation('device_details')
 
   return (
-    //    TODO Immediately: iconColor should be animated
     <>
       <StatusLabel
         status={moduleStatus}
         backgroundColor={C_SKY_BLUE}
         iconColor={C_BLUE}
+        pulse={moduleStatus === 'engaged'}
```
Comment (shlokamin) [inline app/src/organisms/Devices/ModuleCard/MagneticModuleData.tsx:34]: 😄

File: app/src/organisms/Devices/ModuleCard/index.tsx [hunk 7]
```diff
@@ -98,7 +109,7 @@ export const ModuleCard = (props: ModuleCardProps): JSX.Element | null => {
             </Text>
             <Flex paddingBottom={SPACING_1}>
               <ModuleIcon moduleType={module.type} />
-              <Text fontSize={'0.6875rem'}>
+              <Text fontSize={TYPOGRAPHY.fontSizeP}>
```
Comment (shlokamin) [inline app/src/organisms/Devices/ModuleCard/index.tsx:112]: nice!

**Review by shlokamin — APPROVED** [review]
great work Sakib 😄 

just one thing, and we can tackle it later once mel has a chance to give in depth reviews, but it looks like there should be a little more space in between the `target` and `current` temp labels. just noticed the same thing for the temperature module card, but thats not part of this PR

**Review by jerader — APPROVED** [review]
🦖
